### PR TITLE
Feature/ins 184 contacts tableviwe

### PR DESCRIPTION
**Pull request by vierrra**

### CODE REVIEW

File: StantUiIosLibrary/Classes/Headers/FirstHeader/FirstHeaderView.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class FirstHeaderView: UIView {
     }
     
     deinit {
-        self.dropdownMenu   = nil
+        //self.dropdownMenu   = nil
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Headers/FirstHeader/FirstHeaderView.swift]: Remove the comment.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+//  Created by Stant on 19/11/19.
+//  Copyright © 2019 Stant. All rights reserved.
+//
+
+import UIKit
+import StantUiIosLibrary
+
+class ConstructionContactsViewController: UIViewController {
+    
+     let navigationBarHeight: CGFloat = 60
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift:13]: Remove extra space.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    
+    private var firstHeader: FirstHeaderView?
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.view.backgroundColor = .white
+        
+        self.addFirstHeaderView()
+        self.addSecondHeaderView()
+    }
+    
+    fileprivate func addFirstHeaderView() {
+        firstHeader = FirstHeaderView(frame: CGRect(x: 0, y: 0, width: self.view.frame.width, height: self.view.frame.height))
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        
+        let dropdownImage = UIImage(named: "") ?? UIImage()
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        let notificationIcon = UIImage(named: "notifications") ?? UIImage()
+        let dropdownButton = FirstHeaderDropdownButton(title: "", action: #selector(openRecentConstructionSiteDialog), target: self, image: dropdownImage)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: When the line is too big in a function call, try to indent it for the next line.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        let rightBarButton       = FirstHeaderRightBarButton(action: #selector(notificationClick), target: self, image: notificationIcon)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Remove extra spaces to maintain the block indented.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        
+        guard let firstHeader = firstHeader else { return }
+        firstHeader.configure(dropdown: dropdownButton, rightBarButton: rightBarButton)
+        self.view.addSubview(firstHeader)
+        firstHeader.anchor(top: self.view.safeAreaLayoutGuide.topAnchor,
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    @objc func openRecentConstructionSiteDialog() {
+        //TODO: Here it cames dialog with last construction sites.
+    }
+    
+    @objc func notificationClick() {
+           //TODO: Here it cames notification screen.
+    }
+    
+    fileprivate func addSecondHeaderView() {
+        navigationView = SecondHeaderView(frame: CGRect(x: 0, y: 0, width: self.view.frame.width,height: SecondHeaderView.headerHeight))
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift [hunk 10]
```diff
@@ -39,7 +39,6 @@ class FirstHeaderViewController: UIViewController, RecentItemCellDelegate {
         
         let dropdownImage        = UIImage(named: "down") ?? UIImage()
         let notificaticationIcon = UIImage(named: "notifications") ?? UIImage()
-        
         let dropdownButton       = FirstHeaderDropdownButton(title: headerTitle, action: #selector(openRecentItemsDialog), target: self, image: dropdownImage)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift:42]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift [hunk 11]
```diff
@@ -39,7 +39,6 @@ class FirstHeaderViewController: UIViewController, RecentItemCellDelegate {
...
         let rightBarButton       = FirstHeaderRightBarButton(action: #selector(notificationClick), target: self, image: notificaticationIcon)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift:43]: Indentation.

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/Main/MainViewController.swift [hunk 12]
```diff
@@ -135,4 +136,12 @@ class MainViewController: UIViewController {
     @objc func goToSecondHeaderDemo() {
         self.open(viewController: SecondHeaderViewController())
     }
+    
+    fileprivate func addConstructionsContactsDemoTableView() {
+        self.view.addButton(title: "Contruction Contacts Table View", action: #selector(gotToConstructionsContactsDemo), orderOfButton: 13, target: self)
```
Comment (Leandro97) [inline StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/Main/MainViewController.swift:141]: Indentation.

File: StantUiIosLibrary/Classes/Headers/FirstHeader/FirstHeaderView.swift (hunk 1 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Headers/FirstHeader/FirstHeaderView.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift (hunk 2 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/ConstructionsContacts/ConstructionContactsViewControler.swift:13]: Done

**Review by Leandro97 — APPROVED** [review]

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+    
+    public func configureViewFor(construction: ContactsInformation) {
+        self.removeSubviews()
+        
+        self.backgroundColor = .clear
+        
+        self.addMainViewWithShadow()
+        self.configureImageWith(url: construction.imageUrl ?? String(), iconSize: construction.iconSize ?? CGSize())
+        self.configure(name: construction.name ?? String(), office: construction.office ?? String())
+        self.set(mailButton: construction.mailButton, mailImage: construction.mailImage ?? UIImage(), callButton: construction.callButton, callImage: construction.callImage ?? UIImage())
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Indentation.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+    }
+    
+    private func set(mailButton: UIButton? = nil, mailImage: UIImage, callButton: UIButton? = nil, callImage: UIImage) {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break the parameters like:

```
private func set(mailButton: UIButton? = nil,
mailImage:  UIImage,
callButton:  UIButton? = nil,
callImage:   UIImage) {

```

Comment (Leandro97) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Indent it.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+    }
+    
+    fileprivate func configure(name: String, office: String) {
+        nameLabel    = UILabel(frame: CGRect(x: 68, y: 173, width: 210, height: 19))
+        officeLabel  = UILabel(frame: CGRect(x: 68, y: 194, width: 210, height: 14))
+        
+        guard let nameLabel = nameLabel, let officeLabel = officeLabel else { return }
+        
+        if #available(iOS 8.2, *) {
+            self.setText(label: nameLabel, text: name, textSize: 16, textWeight: .regular, textColor: .darkStant,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+                         topAnchor: 5, leftAnchor: 62, bottomAnchor: 26, rightAnchor: 82)
+            self.setText(label: officeLabel, text: office, textSize: 12, textWeight: .regular, textColor: .darkGrayStant,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+                         topAnchor: 28, leftAnchor: 62, bottomAnchor: 10, rightAnchor: 82)
+        }
+    }
+    
+    fileprivate func setText(label: UILabel, text: String, textSize: CGFloat, textWeight: UIFont.Weight, textColor: UIColor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+            label.text      = text
+            label.font      = .systemFont(ofSize: textSize, weight: textWeight)
+            label.textColor = textColor
+            
+            mainView.addSubview(label)
+            label.anchor(top:      mainView.topAnchor,
+                         leading:  mainView.leadingAnchor,
+                         bottom:   mainView.bottomAnchor,
+                         trailing: mainView.trailingAnchor,
+                         padding:  UIEdgeInsets(top: topAnchor, left: leftAnchor, bottom: bottomAnchor, right: rightAnchor))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+        }
+        
+    }
+    
+    fileprivate func configureImageWith(url: String, iconSize: CGSize) {
+        photoImageView = RoundedImageView(frame: CGRect(x: 19, y: 173, width: iconSize.width, height: iconSize.height))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+    let name:     String?
+    let office:   String?
+    let imageUrl: String?
+    var iconSize: CGSize?
+    let mailButton: UIButton?
+    let mailImage: UIImage?
+    let callButton: UIButton?
+    let callImage: UIImage?
+    
+    public init(name: String, office: String, imageUrl: String, iconSize: CGSize, mailButton: UIButton?, mailImage: UIImage, callButton: UIButton?, callImage: UIImage) {
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Break and indent the parameters.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+        
+        guard let searchView = searchView else { return }
+        self.addSubview(searchView)
+        searchView.configureViewWith(delegate:        self,
+                                     image:           searchBarIcon,
+                                     placeholderText: searchBarPlaceholder)
+    }
+    
+    fileprivate func configureTableView() {
+        tableView = ContactsTableView(frame: CGRect(x: 0, y: DefaultSearchBar.searchViewHeight, width: self.frame.width, height: self.frame.height))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Break and indent the parameters on the lines 65, 66 and 67.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+                          leading:  self.leadingAnchor,
+                          bottom:   self.bottomAnchor,
+                          trailing: self.trailingAnchor,
+                          padding:  UIEdgeInsets(top: 0, left: 6, bottom: self.frame.height - DefaultSearchBar.searchViewHeight, right: 6))
+        
+        tableView.anchor(top:      searchView.topAnchor,
+                         leading:  self.leadingAnchor,
+                         bottom:   self.bottomAnchor,
+                         trailing: self.trailingAnchor,
+                         padding: UIEdgeInsets(top: DefaultSearchBar.searchViewHeight, left: 0, bottom: 0, right: 0))
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+        guard let emptyMessageLabel = emptyMessageLabel, let tableView = tableView else { return }
+        emptyMessageLabel.anchor(top:      tableView.topAnchor,
+                                 leading:  self.leadingAnchor,
+                                 trailing: self.trailingAnchor,
+                                 padding:  UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16),
+                                 size:     CGSize(width: 0, height: 50))
+        emptyMessageLabel.isHidden = false
+    }
+}
+
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Remove extra lines.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+
+extension ContactsTableViewWithCollapseSearchBar: ContactsAndHideSearchTableViewDelegate {
+    public func collapseSearchView() {
+        guard let searchView = searchView, let tableView = tableView else { return }
+        self.removeAllConstraints()
+               
+        UIView.animate(withDuration: 0.2, delay: 0, options: [.transitionCrossDissolve], animations: {
+                searchView.isHidden = true
+                   tableView.removeAllConstraints()
+                   tableView.frame     = CGRect(x: tableView.frame.minX, y: 0, width: self.frame.width,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Remove extra spaces and break CGRect parameters.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+                                                height: self.frame.height)
+               }, completion: { _ in
+                   tableView.anchor(top: self.topAnchor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Indent lines 21 to 30.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift [hunk 26]
```diff
@@ -0,0 +1,61 @@
...
+        guard let searchView = searchView, let tableView = tableView else { return }
+        self.removeAllConstraints()
+        
+        UIView.animate(withDuration: 0.2, delay: 0, options: [.transitionCrossDissolve], animations: {
+            searchView.isHidden = false
+            tableView.removeAllConstraints()
+            tableView.frame     = CGRect(x: tableView.frame.minX, y: DefaultSearchBar.searchViewHeight, width: self.frame.width,
+                                         height: self.frame.height - DefaultSearchBar.searchViewHeight)
+        }, completion: { _ in
+            searchView.anchor(top: self.topAnchor,
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Indent all this block.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 27]
```diff
@@ -0,0 +1,113 @@
...
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: ConstructionContacts.identifier(), for: indexPath) as? ConstructionContacts else {
+            return UITableViewCell()
+       }
+        
+        cell.configureViewFor(construction: filteredConstructionSiteList[indexPath.row])
+
+           return cell
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Indentation.

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift [hunk 28]
```diff
@@ -0,0 +1,113 @@
...
+        let absoluteBottom: CGFloat = scrollView.contentSize.height - scrollView.frame.size.height
+            
+        let isScrollingDown = scrollDiff > 0 && scrollView.contentOffset.y > absoluteTop
+        let isScrollingUp   = scrollDiff < 0 && scrollView.contentOffset.y < absoluteBottom
+            
+        if canAnimateHeader(scrollView) {
+            if isScrollingDown {
+                animationDelegate?.collapseSearchView()
+            } else if isScrollingUp {
+                    animationDelegate?.expandSearchView()
```
Comment (Leandro97) [inline StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Indentation,

**Review by Leandro97 — CHANGES_REQUESTED** [review]

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 13 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 14 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 15 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 16 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 17 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 18 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 19 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift (hunk 20 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 21 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 22 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift (hunk 23 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/ContactsTableViewWithCollapsedSearchBar.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift (hunk 24 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift (hunk 25 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift (hunk 26 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/Extensions/ContactsViewShowAndHideSearchTableViewExtension.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 27 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Done

File: StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift (hunk 28 above)
Comment (vierrra) [reply StantUiIosLibrary/Classes/Tables/ContactsTableViewWithCollapsedSearchBar/TableView/ContactsTableView.swift]: Done

File: StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift (hunk 10 above)
Comment (vierrra) [reply StantUiIosLibraryDemo/StantUiIosLibraryDemo/DemoApplication/FirstHeader/FirstHeaderViewController.swift:42]: Done

**Comment by andersonfernandes** [comment]
Review those tests, they failed on Travis. You can see the details of the build by clicking on details.
image

File: StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift [hunk 29]
```diff
@@ -0,0 +1,242 @@
+//
```
Comment (andersonfernandes) [inline StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: You could rename this file and the class name to ContactsCard.

You are creating a new card component here, so the correct place to this file should be in another folder inside the Cards folder (StantUiIosLibrary/Classes/Cards/Contacts/ContactsCard.swift).

Comment (vierrra) [reply StantUiIosLibrary/Classes/Cards/Construction/ConstructionContacts.swift]: Done

**Review by andersonfernandes — APPROVED** [review]

**Pull request closed without merging**